app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorUpdate.kt

```diff
@@ -4,15 +4,78 @@ import com.spotify.mobius.Next
 import com.spotify.mobius.Next.noChange
 import com.spotify.mobius.Update
 import org.simple.clinic.mobius.next
+import org.simple.clinic.sync.LastSyncedState
+import org.simple.clinic.sync.SyncInterval
+import org.simple.clinic.sync.SyncProgress.FAILURE
+import org.simple.clinic.sync.SyncProgress.SUCCESS
+import org.simple.clinic.sync.SyncProgress.SYNCING
+import org.simple.clinic.sync.indicator.SyncIndicatorState.ConnectToSync
+import org.simple.clinic.sync.indicator.SyncIndicatorState.SyncPending
+import org.simple.clinic.sync.indicator.SyncIndicatorState.Synced
+import org.simple.clinic.sync.indicator.SyncIndicatorState.Syncing
+import org.threeten.bp.Duration
+import org.threeten.bp.Instant
+import java.util.concurrent.TimeUnit
 
 class SyncIndicatorUpdate : Update<SyncIndicatorModel, SyncIndicatorEvent, SyncIndicatorEffect> {
   override fun update(model: SyncIndicatorModel, event: SyncIndicatorEvent):
       Next<SyncIndicatorModel, SyncIndicatorEffect> {
 
     return when (event) {
-      is LastSyncedStateFetched -> next(model.lastSyncedStateChanged(event.lastSyncState))
-      is DataForSyncIndicatorStateFetched -> noChange()
-      is IncrementTimerTick -> noChange()
+      is LastSyncedStateFetched -> next(model.lastSyncedStateChanged(event.lastSyncState), FetchDataForSyncIndicatorState)
+      is DataForSyncIndicatorStateFetched -> updateSyncIndicatorState(model, event.currentTime, event.syncIndicatorFailureThreshold)
+      is IncrementTimerTick -> next(model.syncIndicatorSyncedTimerIncremented(event.minutes))
     }
   }
+
+  private fun updateSyncIndicatorState(
+      model: SyncIndicatorModel,
+      currentTime: Instant,
+      failureThreshold: Duration
+  ): Next<SyncIndicatorModel, SyncIndicatorEffect> {
+    if (model.lastSyncedState == null || model.lastSyncedState.isEmpty())
+      return noChange()
```

Comment:
Same comment as earlier about either wrapping this in braces or moving the `return` onto the same line.